device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/ModuleClient.java

```diff
@@ -322,9 +322,15 @@ public static ModuleClient createFromEnvironment(UnixDomainSocketChannel unixDom
 
                 return new ModuleClient(iotHubAuthenticationProvider, protocol);
             }
-            catch (IOException | TransportException | HsmException | URISyntaxException | CertificateException | NoSuchAlgorithmException | KeyStoreException | KeyManagementException e)
+            catch (HsmException | URISyntaxException | CertificateException | NoSuchAlgorithmException | KeyStoreException | KeyManagementException e)
             {
-                throw new ModuleClientException(e);
+                // TODO
```

Comment:
todo?

Reply:
This PR will need some further work after we've agreed on the principles of what we want to do. I have some cases like this and some documentation to iron out still though